### PR TITLE
Confirm modal to revoke/remove from project or org

### LINKED ISSUE #11641: Missing prompt when removing a user from an organization
When a user is removed from an organization, a prompt should appear asking 'Are you sure?'. This prompt is missing at the moment. 

To replicate this issue-

1. Click on 'Your organizations'
2. Click on 'Manage'
3. Scroll to the bottom of the page where the members of the organization are listed
4. Click on 'Remove' against any of the members

A banner appears at the top that the user has been removed but no prompt appears

org1
org2

### LINKED ISSUE #11649: Confirmation message when revoking an invitation
When an owner has invited a user to join an organization, this invitation can be revoked. When the invitation is revoked, there is no confirmation message that appears before proceeding with the operation.

To replicate this issue-

1. Click on 'Your organizations'
2. Click on 'Manage'
3. Scroll to the bottom of the page where the members of the organization are listed
4. Click on 'Revoke Invite' against any of the pending invitations

### PR DESCRIPTION
**Pull request by divbzero**
- Closes #11641.
- Closes #11649.

### CODE REVIEW

**Comment by divbzero** [comment]
Fixed up Stimulus controller binding in commit [3776484](https://github.com/pypi/warehouse/pull/12020/commits/377648483abe2ff328777962bc1a938d449b1bdc). Ready for review.

File: warehouse/templates/manage/organization/roles.html [hunk 1]
```diff
@@ -114,25 +114,16 @@ <h2>{% trans %}People{% endtrans %}</h2>
...
+        {% elif request.has_permission("manage:organization") and request.user.username == role.user.username %}
+          <button class="button" title="{% trans %}Cannot remove yourself as owner{% endtrans %}" disabled>{% trans %}Remove{% endtrans %}</button>
+        {% else %}
+          {% set extra_fields %}<input type="hidden" name="role_id" value="{{ role.id }}">{% endset %}
+          {% set extra_description %}{% trans user=('yourself' if request.user.username == role.user.username else role.user.username) %}Remove {{ user }} from this organization{% endtrans %}{% trans %}.{% endtrans %}{% endset %}
+          {% set tooltip %}{% trans user=('yourself' if request.user.username == role.user.username else role.user.username) %}Remove {{ user }} from this organization{% endtrans %}{% endset %}
+          {{ confirm_button(gettext("Remove"), None, "remove-" + role.id|string, None, extra_fields=extra_fields, extra_description=extra_description, action=request.route_path('manage.organization.delete_role', organization_name=organization.normalized_name), warning=False, modifier="--primary", tooltip=tooltip) }}
```
Comment (ewdurbin) [inline warehouse/templates/manage/organization/roles.html:125]:
```suggestion
          {{ confirm_button(gettext("Remove"), gettext("Username"), "remove-" + role.id|string, gettext(role.user.username), extra_fields=extra_fields, extra_description=extra_description, action=request.route_path('manage.organization.delete_role', organization_name=organization.normalized_name), warning=False, modifier="--primary", tooltip=tooltip) }}
```

File: warehouse/templates/manage/project/roles.html [hunk 2]
```diff
@@ -148,22 +137,14 @@ <h2>{% trans %}Collaborators{% endtrans %}</h2>
...
-            Remove
-            </button>
-          </form>
+        {% if role.role_name == TeamProjectRoleType.Administer and request.user in role.team.members and request.user not in project.organization.owners %}
+          <button type="submit" class="button" title="{% trans %}Cannot remove your own team with Administer permissions{% endtrans %}" disabled>{% trans %}Remove{% endtrans %}</button>
+        {% else %}
+          {% set extra_fields %}<input type="hidden" name="role_id" value="{{ role.id }}">{% endset %}
+          {% set extra_description %}{% trans collaborator=role.team.name %}Remove {{ collaborator }} from this project{% endtrans %}{% trans %}.{% endtrans %}{% endset %}
+          {% set tooltip %}{% trans collaborator=role.team.name %}Remove {{ collaborator }} from this project{% endtrans %}{% endset %}
+          {{ confirm_button(gettext("Remove"), None, "remove-" + role.id|string, None, extra_fields=extra_fields, extra_description=extra_description, action=request.route_path('manage.project.delete_team_project_role', project_name=project.name), warning=False, modifier="--primary", tooltip=tooltip) }}
```
Comment (ewdurbin) [inline warehouse/templates/manage/project/roles.html:146]:
```suggestion
          {{ confirm_button(gettext("Remove"), gettext("Username"), "remove-" + role.id|string, gettext(role.user.username), extra_fields=extra_fields, extra_description=extra_description, action=request.route_path('manage.project.delete_team_project_role', project_name=project.name), warning=False, modifier="--primary", tooltip=tooltip) }}
```

**Review by ewdurbin — APPROVED** [review]
Approved with suggestion that while revoking an invite is fine without further confirmation, we do likely want to add further confirmation when removing an established role.

Users must confirm username to do so.

**Pull request merged**